### PR TITLE
Add Food Show and Update Endpoints

### LINKED ISSUE #1: User can find all foods
User can request `GET /api/v1/foods`, then response returns all foods currently in the database

Response format:
```
{
    "id": 1,
    "name": "Banana",
    "calories": 150
},
```
### Checks
- [x] For success, returns status code 200
- [x] Returns correct response format json

Sad path
- [x] For error, returns status code 500
- [x] Database is empty

### LINKED ISSUE #2: User can find single food
User can request `GET /api/v1/foods/:id`, then response that food from the database

Response format:
```
{
    "id": 1,
    "name": "Banana",
    "calories": 150
}
```
### Checks
- [x] For success, returns status code 200
- [x] Returns correct response format json

Sad path
- [x] Status code 404 if the food is not found

### LINKED ISSUE #6: User can update a food
User can send information 
```
{name": "Banana", "calories": "140"} 
``` 
to `PUT /api/v1/foods/:id` which returns the food item.

Response format:
```
{
    "id": 1,
    "name": "Banana",
    "calories": 140
}
```
### Checks
- [x] For success, returns status code 202
- [x] Returns correct response format json

Sad path
- [x] For error, returns status code 400
- [x] Both name and calories are required fields

### PR DESCRIPTION
**Pull request by lpile**
## What functionality does this accomplish?
**Description:**
* Refactor `app.spec.js` into `tests/` files
* Add edge case test and update route for `GET /api/v1/foods` which closes #1 
* Add tests and route for `GET /api/v1/foods/:id` which closes #2
* Add tests and route for `PUT /api/v1/foods/:id` which closes #6

## What did you struggle on to complete?

* It took awhile to figure out how to test edge cases.

### Test Coverage Percentage: x%
- [ ] No Tests have been changed
- [ ] Some Tests have been changed
- [x] All of the Tests have been changed(Please describe what in the world happened):

## Checklist:
- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code
- [ ] I have partially tested my code (please explain why):

## Helpful Resources:
[Turing Express Tuturial](https://backend.turing.io/module4/lessons/server_side_javascript)
[Mod 4 Jest Lesson](https://backend.turing.io/module4/lessons/jest_testing)

### CODE REVIEW

File: routes/api/v1/foods.js [hunk 1]
```diff
@@ -1,19 +1,86 @@
...
 var Food = require('../../../models').Food;
 
 /* GET all foods */
-router.get("/", function (req, res) {
+router.get('/', function (req, res) {
   Food.findAll()
     .then(foods => {
-      res.setHeader("Content-Type", "application/json");
-      res.status(200).send(JSON.stringify(foods));
+      if (foods.length === 0) {
```
Comment (Mycobee) [inline routes/api/v1/foods.js]:
```suggestion
      if (!array.length) {
```
this is equivalent to array.empty? in ruby

File: tests/requests/foods/index_food.spec.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      expect(Object.keys(response.body[0])).toContain('calories')
+    })
+  });
+
+  test('should return a 404 status if database is empty', () => {
+    Food.destroy({ where: { } })
+
+    return request(app).get('/api/v1/foods').then(response => {
+      expect(response.status).toBe(404)
+      expect(response.body.error).toBe('Database Is Empty')
```
Comment (Mycobee) [inline tests/requests/foods/index_food.spec.js]: I didn't realize you could validate an error like this.  Looks good!

File: routes/api/v1/foods.js (hunk 1 above)
Comment (lpile) [reply routes/api/v1/foods.js]: I like it!

File: tests/requests/foods/index_food.spec.js (hunk 2 above)
Comment (lpile) [reply tests/requests/foods/index_food.spec.js]: Yeah, I found that last night.
